### PR TITLE
Second Order Kinematics

### PR DESCRIPTION
**Pull request by Hi30MC**
See other PR for desc.

### CODE REVIEW

**Comment by Hi30MC** [comment]
Forgot to add param description: 
@param robotTheta Theta Velocity of the robot

should be included in getVelXYFromAccelXY, calculateModulesPositionAccelYMetersPerSecondSquared, and calculateModulesPositionAccelXMetersPerSecondSquared

File: src/main/java/frc/robot/util/Pigeon2Accelerometer.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  private static double fixedToDouble(short fixed) {
+    /** takes raw short and spits out a double Gs (gravitational constant on the surface of earth)
+     * first take short and force it to double type
+     * divide by magic number (pigeon 2 returns Gs in Q2.14 notation)
+     * Q2.14 notation dictates such format: 0000000000000000 -> 00.00000000000000
+     * therefore to get Gs we need to shift the decimal 14 bits to the left
+     * operation for this is to divide by 2^14 hence the divided by a 1 bit shifted to the left 14 times
+    */
+    return ((double) fixed) / (1 << 14);
```
Comment (hadley31) [inline src/main/java/frc/robot/util/Pigeon2Accelerometer.java:95]: Why remove the `kShift`?

Comment (Hi30MC) [reply src/main/java/frc/robot/util/Pigeon2Accelerometer.java:95]: I think that this file was from ages ago when you first made it; if you have another copy of that file with more updated code, use that over the one in my PR

File: src/main/java/frc/robot/util/SwerveModuleAcceleration.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package frc.robot.util;
+
+import java.util.Objects;
+
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+
+public class SwerveModuleAcceleration implements Comparable<SwerveModuleAcceleration> {
+  public double accelMetersPerSecondSquared = 0;
+  private static SwerveModuleState[] states = new SwerveModuleState[4];
```
Comment (hadley31) [inline src/main/java/frc/robot/util/SwerveModuleAcceleration.java]: This should be a non-static, non-array state. You should have two here, `m_previousState` and `m_currentState`. You also need a `m_previousTime` field to properly calculate the acceleration.

Comment (hadley31) [reply src/main/java/frc/robot/util/SwerveModuleAcceleration.java]: Unless we want to keep this as a POD class, then we should remove this entirely, and add a `public final Rotation2d m_angle` field

File: src/main/java/frc/robot/util/SwerveModuleAcceleration.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    return Double.compare(this.accelMetersPerSecondSquared, other.accelMetersPerSecondSquared);
+  }
+
+  @Override
+  public String toString() {
+    return String.format(
+        "SwerveModuleAccel(Accel: %.2f m)", accelMetersPerSecondSquared);
+  }
+
+  private static SwerveModuleAcceleration calculate(SwerveModuleState current, SwerveModuleState previous) {
```
Comment (hadley31) [inline src/main/java/frc/robot/util/SwerveModuleAcceleration.java]: This should take in either a delta time argument, or a timestamp used to calculate the acceleration

Comment (Hi30MC) [reply src/main/java/frc/robot/util/SwerveModuleAcceleration.java]: Fixed

File: src/main/java/frc/robot/util/SwerveModuleAcceleration.java (hunk 2 above)
Comment (Hi30MC) [reply src/main/java/frc/robot/util/SwerveModuleAcceleration.java]: m_previousTime has been replaced by including a deltaTime param in SMA.calculate()
m_previousState and m_currState have been added in replacement of the list
m_angle is omitted as module rotational accel is never used in SOK.calculateModulesPositionAccelX or Y

File: src/main/java/frc/robot/util/Pigeon2Accelerometer.java (hunk 1 above)
Comment (Hi30MC) [reply src/main/java/frc/robot/util/Pigeon2Accelerometer.java:95]: Hasn't been touched in recent commit as I have no idea how you would want to implement a kShift, let me know how you would want to do it.

Comment (Hi30MC) [reply src/main/java/frc/robot/util/Pigeon2Accelerometer.java:95]: Oh, just realized that you made another method that takes in kShift under this, resolving now

**Pull request closed without merging**